### ISSUE TITLE
Google maps API key is exposed

### ISSUE THREAD

**Issue by kanandam** [shukerullah/react-geocode#49]
Hi there, 
I'm using this library for geocoding purpose in my react code, and the API key is exposed in the request under query params, is there a way to encrypt or hide it in the request?

<img width="458" alt="image" src="https://github.com/shukerullah/react-geocode/assets/109073993/d0270dd8-18ca-45e3-85ed-acc215ce2466">

TIA...

**Comment by ziyaddin** [comment]
Hi, it's impossible and it has been a known issue for a long time. Community recommends to create and use a separate API key for each type of operations (e.g. one for geocoding, one for map rendering, etc.) to at least decrease the negative impact of only one key being abused with all types of operations by a malicious user.

**Comment by kanandam** [comment]
Hey @ziyaddin , I've been using this other library for google maps component, it uses some methods to create a security layer for the request api, just letting you know if it can help.
https://github.com/JustFly1984/react-google-maps-api/tree/master/packages/react-google-maps-api


cheers...

**Comment by faure-levoux** [comment]
> Hi there, I'm using this library for geocoding purpose in my react code, and the API key is exposed in the request under query params, is there a way to encrypt or hide it in the request?
> <img alt="image" width="458" src="https://private-user-images.githubusercontent.com/109073993/298874286-d0270dd8-18ca-45e3-85ed-acc215ce2466.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2MDYwMjgsIm5iZiI6MTcwNjYwNTcyOCwicGF0aCI6Ii8xMDkwNzM5OTMvMjk4ODc0Mjg2LWQwMjcwZGQ4LTE4Y2EtNDVlMy04NWVkLWFjYzIxNWNlMjQ2Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEzMFQwOTA4NDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZDg4MjdlMjJmNzA4MTczNDIyNDIzZTExNTY4OWJjN2JkOWM4YjFiNjU1ZjU4ODdjOTVmMmI2OTVhNmU4ZjAxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.nMzaDj0b9NvMFBPROXSZh5sxH6fJIK3FNpgCE4rGX3g">
> 
> TIA...

Hi, 
It's not a problem to expose your api, you simply just restrict your apis from ip address or url website ...
[https://cloud.google.com/api-keys/docs/add-restrictions-api-keys](https://cloud.google.com/api-keys/docs/add-restrictions-api-keys)

**Comment by kanandam** [comment]
@Aupire yes, I'm aware of restricting the API, but we don't have the access to the GCP to restrict, it's with client.
